### PR TITLE
fixes holoparasites sending messages after they have been reset

### LINKED ISSUE #45812: Replaced Holoparasites can send messages they started to send to their host before being replaced
Be a holoparasite, begin to communicate a message to your host. Have the host reroll holoparasites and finally send the message after you are replaced. The message still goes through and it even is shown to be said by the new holoparasite rather than by your old name

Issue reported from  Round ID: 115506 (/tg/Station Sybil [ENGLISH] [US-WEST] [100% LAG FREE])

Reporting client version: 512.1471

### PR DESCRIPTION
**Pull request by tralezab**
fixes #45812

### CODE REVIEW

**Comment by tralezab** [comment]
hold up, something's not right with this fix.

**Comment by ExcessiveUseOfCobblestone** [comment]
is `src` needed?

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -356,8 +356,9 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 
 /mob/living/simple_animal/hostile/guardian/proc/Communicate()
 	if(summoner)
+		var/sender_ckey = src.key
 		var/input = stripped_input(src, "Please enter a message to tell your summoner.", "Guardian", "")
-		if(!input)
+		if(sender_ckey != src.key || !input) //guardian got reset, or did not enter anything
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: remove src from src.key

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 2]
```diff
@@ -356,8 +356,9 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 
 /mob/living/simple_animal/hostile/guardian/proc/Communicate()
 	if(summoner)
+		var/sender_ckey = src.key
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: .

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by tralezab** [comment]
review handled

**Comment by AnturK** [comment]
If you're using key don't call variable ckey please.

**Comment by tralezab** [comment]
makes sense to me! i'll change it

**Pull request merged**